lib/Parse/ParseLiteral.cpp

```diff
@@ -142,6 +142,49 @@ std::string Parser::hexUnquote(llvm::StringRef Quoted) {
   return Result;
 }
 
+std::string Parser::simplifyIntegerLiteral(llvm::StringRef Literal) {
+  std::string LiteralStr = Literal.str();
+  LiteralStr.erase(remove(LiteralStr.begin(), LiteralStr.end(), '_'),
+                   LiteralStr.end());
+
+  // Hex format integer literal
+  if (LiteralStr.size() > 2 && LiteralStr[0] == '0' && LiteralStr[1] == 'x')
+    return LiteralStr;
+  size_t eIndex = 0, dotIndex = 0;
+  int eNum = 0, dotNum = 0;
+  for (size_t i = 0; i < LiteralStr.size(); ++i) {
+    if (LiteralStr[i] == 'e') {
+      ++eNum;
+      eIndex = i;
+    }
+    if (LiteralStr[i] == '.') {
+      ++dotNum;
+      dotIndex = i;
+    }
+  }
+
+  /* Invalid cases: 1e2e, 3e, 1.1.1, 1e2.1 */
+  if (eNum != 1 || LiteralStr.back() == 'e' || dotNum > 1 ||
+      (dotNum == 1 && eIndex < dotIndex))
+    return LiteralStr;
+
+  std::string L = LiteralStr.substr(0, eIndex);
+  std::string R = LiteralStr.substr(eIndex + 1);
+  size_t exponent = std::stoul(R);
```

Comment:
Variable names should be nouns (as they represent state). The name should be camel case, and start with an upper case letter (e.g. Leader or Boats).